guppylang/error.py

```diff
@@ -56,9 +53,13 @@ def ipython_excepthook(
         ) -> Any:
             return hook(etype, value, tb)
 
-        ipython_shell.set_custom_exc((GuppyError,), ipython_excepthook)
+        old_hook = ipython_shell.CustomTB
+        old_exc_tuple = ipython_shell.custom_exceptions
+        ipython_shell.set_custom_exc((Exception,), ipython_excepthook)
         yield
-        ipython_shell.set_custom_exc((), None)
```

Comment:
Resetting the except_hook to `None` falls back to default. This no longer enough since we need to restore the previous hook if we're switching back and forth between the tracing and regular checking context.

Instead we need to remember to previous hook and set it back